Plugins/Detray/include/Acts/Plugins/Detray/DetrayConverter.hpp

```diff
@@ -0,0 +1,146 @@
+
+/// Write the detector to json output
+///
+/// @param dDetector is the detray detector (converted)
+/// @param names a name map for the detector volumes
+void writeToJson(const DetrayDetector& dDetector,
+                 const typename DetrayDetector::name_map& names = {});
+
+/// Conversion method for transform objects to detray::transform payloads
+///
+/// @param t the transform to be converted
+///
+/// @return the transform_payload(translation, rotation)
+io::transform_payload convertTransform(const Transform3& t);
+
+/// Conversion method for surface bounds to detray::mask payloads
+///
+/// @param bounds the bounds object to be converted
+/// @param portal the flag for conversion into detray portal format
+///
+/// @return the mask_payload representing the bounds
+io::mask_payload convertMask(const SurfaceBounds& bounds, bool portal = false);
+
+/// Conversion method for surface objects to detray::surface payloads
+///
+/// @param gctx the geometry context
+/// @param surface the surface to be converted
+/// @param portal the flag for conversion into detray portal format
+///
+/// @return the surface_payload for portals and volumes by @param Surface acts object
+io::surface_payload convertSurface(const GeometryContext& gctx,
+                                   const Surface& surface, bool portal = false);
+/// Conversion method for Portal object to detray::portal payloads
+///
+/// @param gctx the geometry context
+/// @param portal the portal to be converted
+/// @param ip the portal index
+/// @param volume the volume to which the portal belongs
+/// @param orientedSurfaces the oriented surfaces of the portal
+/// @param detectorVolumes the detector volumes for the link lookup
+///
+/// @note due to portal splitting this can add up in N portals for one initial one
+///
+/// @brief convert the acts portal to detray surface payload and populate the payload
+std::vector<io::surface_payload> convertPortal(
+    const GeometryContext& gctx, const Experimental::Portal& portal,
+    std::size_t ip, const Experimental::DetectorVolume& volume,
+    const std::vector<OrientedSurface>& orientedSurfaces,
+    const std::vector<const Experimental::DetectorVolume*>& detectorVolumes);
+
+/// Conversion method for volume objects to detray::volume payloads
+///
+/// @param gctx the geometry context
+/// @param volume the volume to be converted
+/// @param detectorVolumes the detector volumes for the link lookup
+///
+/// @return the volume_payload for portals and volumes by @param volume acts object
+io::volume_payload convertVolume(
+    const GeometryContext& gctx, const Experimental::DetectorVolume& volume,
+    const std::vector<const Experimental::DetectorVolume*>& detectorVolumes);
+
+/// Conversion method for (common) header payload
+///
+/// @param detector is the detector to be converted
+///
+/// @return a geometry header payload
+io::geo_header_payload convertHead(
+    const Acts::Experimental::Detector& detector);
```

Comment:
Is this used for ```toDetrayJson``` or why do we need this? The header should only be needed for file IO not for direct conversion